vm/vm.go

```diff
@@ -102,6 +102,69 @@ func matchesAssetParams(newAsset, prevAsset *asset.Asset,
 	return nil
 }
 
+// ValidateVPsbt is a helper method that checks a vPSBT for validity. This
+// includes checking the provided signatures against the fields of the transfer
+// they commit to.
+func ValidateVPsbt(newAsset *asset.Asset, splitAssets []*commitment.SplitAsset,
+	prevAssets commitment.InputSet) error {
```

Comment:
Due to a circular dependency issue that would eventually be introducing by merging both this PR and https://github.com/lightninglabs/taproot-assets/pull/787 on `main`, this func was changed to not accept a `Vpkt` as that would import `taproot-assets/tappsbt`, which would be the source of the circular dep. issue

Reply:
Since this isn't a vPSBT we're validating anymore, maybe we should rename the interface and method. And while we're at it, give it a more descriptive name. Otherwise what's the difference in functionality between the `TxValidator` and the `VPsbtValidator`?
One verifies the finished transaction in a proof while the other one validates a (partial) signature that was just created. So maybe `SigningValidator` or something like that would be more descriptive?

Reply:
Also maybe extending the comment a bit explaining _why_ exactly this is different from the `TxValidator` would probably be good. For example that in case of a multi-signature we can't fully verify the full witness, only part of it (for which we might add a TODO in the code somewhere).